docs/developer-docs/getting-started/write-smart-contracts.mdx

```diff
@@ -0,0 +1,624 @@
+
+:::
+
+### 1. Open the backend canister source code file in your code editor.
+
+For Motoko projects, this file will be `src/PROJECT_NAME_backend/main.mo`.
+
+For Rust projects, this file will be `src/PROJECT_NAME_backend/src/lib.rs`.
+
+For [Azle](https://demergent-labs.github.io/azle/get_started.html) or [Kybra](https://demergent-labs.github.io/kybra/) projects, please refer to their respective documentation for details about their default code files.
+
+If you are using the default template created by `dfx`, you will see the following default "Hello, world!" code:
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko title="src/PROJECT_NAME_backend/main.mo"
+actor {
+public query func greet(name : Text) : async Text {
+    return "Hello, " # name # "!";
+  };
+};
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust title="src/PROJECT_NAME_backend/src/lib.rs"
+#[ic_cdk::query]
+fn greet(name: String) -> String {
+    format!("Hello, {}!", name)
+}
+```
+
+</TabItem>
+</AdornedTabs>
+
+:::danger
+
+If you are using an ICP Ninja project, this code will vary based on the example you downloaded.
+
+:::
+
+### 2. Add randomness.
+
+A key feature of ICP is its ability to generate [onchain randomness](/docs/current/developer-docs/smart-contracts/advanced-features/randomness) using a verifiable random function (VRF) and [chain-key cryptography](https://internetcomputer.org/how-it-works/chain-key-technology). In each round of consensus on a subnet, the VRF is evaluated using the number of the round as input, producing a fresh set of random bytes. Then, the bytes are used as the seed for a pseudorandom number generator (PRNG), called random tape, which uses chain-key cryptography to create a random, unique value for each canister that requests it. Through this process, it is impossible to predict future outputs.
+
+To use randomness in your canister, you need to make a call to the `raw_rand` method of the [management canister](/docs/current/developer-docs/smart-contracts/advanced-features/management-canister).
+
+To demonstrate how to use randomness, you will create a simple app that generates a random number.
+
+Remove the existing code in your backend source code file. Then, insert the following:
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko title="src/PROJECT_NAME_backend/main.mo"
+import Timer "mo:base/Timer";
+import Nat8 "mo:base/Nat8";
+import Debug "mo:base/Debug";
+import Blob "mo:base/Blob";
+import Nat "mo:base/Nat";
+
+actor {
+    // Create a stable variable to store the current random number.
+    // Stable variables persist across canister upgrades.
+    stable var currentRandomNumber : Nat = 0;
+
+    // The management canister's principal ID is "aaaaa-aa".
+    // This code provides the logic to call the raw_rand method of the management canister.
+    let CallRand : actor {
```

Comment:
I would generally prefer if we update all the docs in that regard to use following import:

```motoko
import IC "ic:aaaaa-aa";
```

we can then directly use and functions and access types of the management canister without having to define `raw_rand()` again:

```motoko
await IC.raw_rand();
```

it would be great to keep this consistent in all places/examples. but we should also make sure then that usage of this is as smooth as possible in `dfx` and the VScode plugin. (cc @jasonaibrahim)